src/components/create-from-builder/create-from-builder.controller.ts

```diff
@@ -233,4 +224,192 @@ export class CreateFromBuilderController implements angular.IController {
     let name = this.ctrl.imageStream.resource.metadata.name + ":" + this.ctrl.istag.name;
     return this.DataService.get("imagestreamtags", name, { namespace: 'openshift' });
   }
+
+  private preselectService() {
+    var newestReady: any;
+    var newestNotReady: any;
+    var statusCondition: any = this.$filter('statusCondition');
+
+    _.each(this.ctrl.serviceInstances, function(instance: any) {
+      var ready = _.get(statusCondition(instance, 'Ready'), 'status') === 'True';
+      if (ready && (!newestReady || instance.metadata.creationTimestamp > newestReady.metadata.creationTimestamp)) {
+        newestReady = instance;
+      }
+      if (!ready && (!newestNotReady || instance.metadata.creationTimestamp > newestNotReady.metadata.creationTimestamp)) {
+        newestNotReady = instance;
+      }
+    });
+    this.ctrl.serviceToBind = _.get(newestReady, 'metadata.name') || _.get(newestNotReady, 'metadata.name');
+  };
+
+  private sortServiceInstances() {
+    if (this.ctrl.serviceInstances) {
+      var instances = _.toArray(this.ctrl.serviceInstances);
+      var statusCondition: any = this.$filter('statusCondition');
+
+      instances.sort(function(left: any, right: any) {
+        var leftReady: boolean = _.get(statusCondition(left, 'Ready'), 'status') === 'True';
+        var rightReady: boolean = _.get(statusCondition(right, 'Ready'), 'status') === 'True';
+
+        if (leftReady === rightReady) {
+          var leftCreated: any = _.get(left, 'metadata.creationTimestamp');
+          var rightCreated: any = _.get(right, 'metadata.creationTimestamp');
+          return  rightCreated.localeCompare(leftCreated);
+        } else {
+          return leftReady ? -1 : 1;
+        }
+      });
+      this.ctrl.serviceInstances = instances;
+    }
+  };
+
+  private updateSteps() {
```

Comment:
We set the title when we enter the page, updateSteps is slightly misnamed as it only occurs when the project selection changes. I'll try to simplify.